src/main/java/core/basesyntax/strategy/ReturnOperation.java

```
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.db.Storage;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        Storage.storage.merge(fruitTransaction.getFruit(),
+                fruitTransaction.getQuantity(), Integer::sum);
```

When throwing errors for invalid input (null transaction, invalid fruit, negative quantity), rethrow informative RuntimeExceptions rather than allowing unchecked exceptions to bubble up. The checklist requires informative error messages and rethrowing RuntimeException with cause when appropriate ("Don't ignore exceptions in try-catch blocks ... Let's rethrow a RuntimeException with an **informative** message and exception object."). Apply this to all validation failures in this handler (apply at line 8 and merge at lines 9–10).